### PR TITLE
Add datepickers to the Announcements forms

### LINKED ISSUE #356: Announcements Form needs new Datepickers

### PR DESCRIPTION
**Pull request by jez**
Commit Summary
--------------

2c0aacf
- Strip whitespace

26c10e0
- Make datetimepicker support code more robust

58d09a4
- Use FormBuilder for datetimepickers. Fixes #356.

fd7558f
- Move 'confirm changes' script to dedicated file.

### CODE REVIEW

**Comment by dlbucci** [comment]
The dates are initially blank when I load the page, and stay that way until I click them (although they appear to work).
Also the persistent announcements are shown on every page in a course, vs only assessments#index, so the description is wrong.

**Comment by jez** [comment]
The dates are initially blank for me too (the form used to be that way too).

 I changed the description, so this should be good to go.

**Comment by dlbucci** [comment]
The old announcement form, you mean?  Is there a way we can fix it?

**Comment by jez** [comment]
What would you prefer? It's a form for creating a new announcement. There's nothing we could pre-populate it with except placeholder data, but then they'd just have to delete it.

**Comment by dlbucci** [comment]
Okay, do they show up for the editing form? And does it work if you never
touch them?

**Comment by jez** [comment]
Oh, it looks like it doesn't on the Edit form. I'll see if I can figure it out. My initial guess is that it's just not saving the field on submit.

**Comment by jez** [comment]
This should be fixed in 364858f.

File: app/views/announcements/new.html.erb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 <h2>Create Announcement</h2>
-<%= form_for :announcement, url:{action:"create"} do |f| %>
-  <%= render partial:'announcementFields', locals:{f:f} %>
-<% end %>
+<div class="row">
+  <div class="col-md-6">
+    <%= form_for @announcement, url:{action:"create"}, builder: FormBuilderWithDateTimeInput do |f| %>
```
Comment (dlbucci) [inline app/views/announcements/new.html.erb]: This line causes an error for new announcements.  The announcement variable needs to be initialized in the controller.

**Comment by dlbucci** [comment]
Did you fix the create announcement error?

**Comment by jez** [comment]
Yeah, but I remade the PR because of the merge conflicts.

**Pull request closed without merging**